parsers/IN_MH.py

```diff
@@ -0,0 +1,386 @@
+
+        image = imread(url)
+        image = Image.fromarray(image)  # create PIL image
+        #image = Image.open('error.png')
+
+        line = ''
+        labels = {}
+        values = {}
+
+        localtime = arrow.utcnow().shift(hours=5, minutes=30)
+        localtime = localtime.format('YYYY-MM-DDTHH:mm')
+        filename_time = localtime.replace(':', ' ')
+
+        plt_num = 1
+        fig = plt.figure(figsize=(3,23))
+        plt.subplots_adjust(top=0.8, wspace=0.2, hspace=0.3)
+        rows = len(locations)
+        cols = 2
+
+        #recognize label and value for all items in locations-dict
+        for type, locs in locations.items():
+            label, l_img = recognize(locs['label'], image, 'eng')
+            value, v_img = recognize(locs['value'], image, 'digits_comma')
+            labels[type] = label
+            values[type] = float(value)
+
+            axes = fig.add_subplot(rows, cols, plt_num)
+            axes.get_xaxis().set_visible(False)
+            axes.get_yaxis().set_visible(False)
+            plt.imshow(l_img)
+            plt.title(label)
+            plt_num = plt_num + 1
+
+            axes = fig.add_subplot(rows, cols, plt_num)
+            axes.get_xaxis().set_visible(False)
+            axes.get_yaxis().set_visible(False)
+            plt.imshow(v_img)
+            plt.title(value)
+            plt_num = plt_num + 1
+
+        share = values['CS EXCH'] / values['CS GEN. TTL.']
+        for type, plants in generation_map.items():
+            for plant in plants['add']:
+                fac = share if plant in CS else 1  # add only a fraction of central state plant production
+                results[type] += fac * values[plant]
+            for plant in plants['subtract']:
+                fac = share if plant in CS else 1
+                results[type] -= fac * values[plant]
+
+        #create line for log.txt
+        #compare recognized label with name in locations dict to detect errors
+        for key in locations.keys():
+            line = line+str(values[key])+ ' '
+            if not areEqual( labels[key], key ):
+                image.save('error_'+filename_time+'.png')
+                print('Error: ' + labels[key] + ', ' +key)
+
+        demand_diff = sum(results.values()) - values['DEMAND']
+        if (abs(demand_diff) > 5):
+            image.save('error_' + filename_time + '.png')
+            print('Error: Demand Difference = {}'.format(demand_diff))
+
+
+
+        for value in results.values():
+            line = line + str(value) + ' '
+
+        #read daytime from image
+        rec_time, img = recognize( (355,110,524,150), image, 'eng')
+        rec_time = rec_time.replace(' ', 'T')
+
+
+        line = line + localtime +' '
+        line = line + rec_time
+        file.write(line+'\n')
+        print(line)
+        #print(results)
+        plt.savefig('figures/' + filename_time + '.png')
+        #plt.show()
+        time.sleep(60*10)
+        plt.close()
+
+    file.close()
```

Comment:
We can remove all of this and just have `print(fetch_production())`

Reply:
Ok! 
Should I post this code to the corresponding issue then? Without visualization anyone trying to debug this will have a really hard time.

Reply:
Good idea, you've made me think of an interesting possibility (will open a new issue).